src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -646,6 +647,12 @@ void TemporalMemory::compute(const SDR &activeColumns, bool learn,
 void TemporalMemory::compute(const SDR &activeColumns, bool learn) {
   activateDendrites(learn);
   activateCells(activeColumns, learn);
+  //anomaly computation
+  anomaly_.currentActiveColumns = this->cellsToColumns(activeCells_); //FIXME cellsToColumns from PR#397 
+  anomaly_.previouslyPredictedColumns = this->cellsToColumns(getPredictiveCells()); //TODO move to T-1
+  anomaly_.score = nupic::algorithms::anomaly::computeRawAnomalyScore(
```

Comment:
should I move the method to TM too? I'm inclining to keeping it as a static method in Anomaly.hpp

Reply:
> I'm inclining to keeping it as a static method in Anomaly.hpp

That's fine.  It's easier to test if its separate too.